### PR TITLE
Fix the use of scheduler pools in BinderServer.

### PR DESCRIPTION
**Pull request by markb74**
Addressing comment from #8031.

### CODE REVIEW

File: binder/src/androidTest/java/io/grpc/binder/internal/BinderTransportTest.java [hunk 1]
```diff
@@ -48,8 +50,8 @@
 public final class BinderTransportTest extends AbstractTransportTest {
 
   private final Context appContext = ApplicationProvider.getApplicationContext();
-  private final ScheduledExecutorService scheduledExecutorService =
-      Executors.newScheduledThreadPool(2);
+  private final ObjectPool<ScheduledExecutorService> executorServicePool =
+      new FixedObjectPool<>(Executors.newScheduledThreadPool(2));
```
Comment (ejona86) [inline binder/src/androidTest/java/io/grpc/binder/internal/BinderTransportTest.java:54]: This is actually (already) creating a lot of threads. I don't know if they are GCed on Android either (in Java 8 I think they are, but in 9+ they aren't in openjdk). You probably want to make this `static` and/or shut it down in `tearDown()`. That can be a follow-up.

Another option is to use `SharedResourcePool.forResource(GrpcUtil.TIMER_SERVICE)`.

File: binder/src/main/java/io/grpc/binder/internal/BinderServer.java [hunk 2]
```diff
@@ -119,14 +121,11 @@ public SocketAddress getListenSocketAddress() {
...
-      if (useSharedTimer) {
-        // TODO: Transports may still be using this resource. They should 
-        // be managing its use as well.
-        SharedResourceHolder.release(GrpcUtil.TIMER_SERVICE, executorService);
-      }
       // Break the connection to the binder. We'll receive no more transactions.
       hostServiceBinder.detach();
       listener.serverShutdown();
+      executorServicePool.returnObject(executorService);
+      executorService = null;
```
Comment (ejona86) [inline binder/src/main/java/io/grpc/binder/internal/BinderServer.java:128]: FYI, returnObject always returns `null` for convenience to allow `executorService = executorServicePool.returnObject(executorService);`. But there's nothing wrong with this.

**Review by ejona86 — APPROVED** [review]

File: binder/src/androidTest/java/io/grpc/binder/internal/BinderTransportTest.java (hunk 1 above)
Comment (markb74) [reply binder/src/androidTest/java/io/grpc/binder/internal/BinderTransportTest.java:54]: Done.

File: binder/src/main/java/io/grpc/binder/internal/BinderServer.java (hunk 2 above)
Comment (markb74) [reply binder/src/main/java/io/grpc/binder/internal/BinderServer.java]: Done.
